systemtest/src/main/java/io/strimzi/systemtest/utils/RollingUpdateUtils.java

```diff
@@ -112,8 +113,16 @@ public static void waitForComponentAndPodsReady(String namespaceName, LabelSelec
         PodUtils.waitForPodsReady(namespaceName, selector, expectedPods, true,
             () -> ResourceManager.logCurrentResourceStatus(KafkaResource.kafkaClient().inNamespace(namespaceName).withName(clusterName).get()));
 
-        KafkaUtils.waitForKafkaReady(namespaceName, clusterName);
-        LOGGER.info("Kafka: {}/{} is ready", namespaceName, clusterName);
+        if (KafkaResource.kafkaClient().inNamespace(namespaceName).withName(clusterName).get() != null) {
+            KafkaUtils.waitForKafkaReady(namespaceName, clusterName);
+            LOGGER.info("Kafka: {}/{} is ready", namespaceName, clusterName);
+        } else {
+            KafkaList kafkaList = KafkaResource.kafkaClient().inNamespace(namespaceName).list();
+            kafkaList.getItems().forEach(kafka -> {
+                KafkaUtils.waitForKafkaReady(kafka.getMetadata().getNamespace(), kafka.getMetadata().getName());
+                LOGGER.info("Kafka: {}/{} is ready", namespaceName, clusterName);
+            });
+        }
```

Comment:
Good point

Reply:
It needs to be also for `KafkaList` because in the edge cases where one changed f.e. MM2 instance, we have to wait on the readiness of Kafka clusters (source and target). Also, I have observed during my testing that sometimes when one does not specify the same `clusterName` as for KafkaConnect or KafkaMirrorMaker, it will throw an NPE. 

For example, one deploys `Kafka` with name (`kafka-name-cluster`) and KafkaMirrorMaker with clusterName = `my-cluster-` and then we will call the `waitForComponentAndPodsReady` method for KafkaMirrorMaker. What happens next is that this method will expect that KafkaCluster has a similar name to KafkaMIrrorMaker, which is not true and will throw NPE.

Reply:
Updated method.